### ISSUE TITLE
replace pattern matching behaving strange

### ISSUE THREAD

**Issue by mfiano** [beetbox/beets#223]
My library has a couple of artists that beets stores in directories based on their artist name and special characters replaced with an underscore.

.38 Special - stored in albums/_38 Special/...
:wumpscut: - stored in albums/_wumpscut_/...

My config.yaml file does not override the 'replace' patterns option and this just works. However, if I copy the patterns from config_default.yaml into my config and issue a 'beet mv', the directory structure is changed to that of:

.38 Special - stored in albums/.38 Special/... (a hidden dot folder on Linux)
:wumpscut: - stored in albums/:wumpscut:/... (problematic characters on Windows)

In other words, if I use replace explicitly in my config, whether changed from the default values or not, the replacements do not happen. However, if I do not include 'replace' patterns in my config, the default works as expected.
